runtime/src/iree-amd-aie/driver/xrt/direct_command_buffer.cc

```diff
@@ -348,8 +347,11 @@ static iree_status_t iree_hal_xrt_direct_command_buffer_dispatch(
         xrt::bo(*command_buffer->descriptor_sets[set].bindings[j],
                 command_buffer->descriptor_sets[set].lengths[j],
                 command_buffer->descriptor_sets[set].offsets[j]);
-    bos.push_back(arg_buffer);
     run.set_arg(arg_index + j, arg_buffer);
+    bool not_ofm = (bindings.values[j].buffer->memory_type & IREE_HAL_MEMORY_TYPE_HOST_VISIBLE) &&
+                   (bindings.values[j].buffer->allowed_usage & IREE_HAL_BUFFER_USAGE_MAPPING);
```

Comment:
`IREE_HAL_BUFFER_USAGE_MAPPING` is also not the correct flag: 

```
  // WARNING: mapping can be extremely expensive, use limited hardware
  // resources, introduce data hazards, and synchronize host and device
  // execution. Unless an application knows that such issues will not arise
  // (as in tests where there's never concurrent usage) mapping should be used
  // judiciously: do not assume mapping is a high-performance technique!
```

https://github.com/iree-org/iree/blob/66342abbfaaee707e27ecc7d8151ad9e357ca0da/runtime/src/iree/hal/buffer.h#L389

Reply:
Please refer to this comment:
https://github.com/nod-ai/iree-amd-aie/pull/849#discussion_r1807139120